examples/hello-triangle/main.rs

```diff
@@ -83,21 +81,13 @@ async fn run(event_loop: EventLoop<()>, window: Window, swapchain_format: wgpu::
     let mut swap_chain = device.create_swap_chain(&surface, &sc_desc);
 
     event_loop.run(move |event, _, control_flow| {
-        // force ownership by the closure
```

Comment:
I've removed this and it seems everything still worked fine

Reply:
The problem is that `run` never actually returns, so things aren't properly dropped. It's not necessary (not calling drop is always safe), but useful to have.

Reply:
As far as I understand, since we set `*control_flow = ControlFlow::Exit` when the user closes the window, `event_loop.run` will stop, and then the big `run` will return?

Reply:
Due to restrictions on the web and mobile, `run` never returns (returns `!`). There is `run_return`, but it's non-portable.

Reply:
Not dropping the instance and stuff means API validation errors on exit. We don't want that. Let's bring back this ownership hack, until we have a better solution

Reply:
Thanks for the clarifications! I've added the code back, with a comment explaining why it's necessary.